src/Utilities/Compiler/Extensions/CompilationExtensions.cs

```diff
@@ -59,5 +61,15 @@ internal static bool TargetsDotNetFramework(this Compilation compilation)
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a value indicating whether the project of the compilation is a Web SDK project,
+        /// based on MSBuild property <see cref="MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb"/>.
+        /// </summary>
+        internal static bool IsWebProject(this Compilation compilation, AnalyzerOptions options, CancellationToken cancellationToken)
+        {
+            var usingMicrosoftNETSdkWeb = options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb, compilation, cancellationToken);
+            return usingMicrosoftNETSdkWeb != null && bool.TryParse(usingMicrosoftNETSdkWeb, out var result) && result;
```

Comment:
Do you need to potentially trim the string option before doing a `TryParse`? Also as you are then checking the fact this is true, isn't it better to replace the full line with `string.Equals(usingMicrosoftNETSdkWeb?.Trim(), "true", StringComparison.OrdinalIgnoreCase)`?